src/Microsoft.Azure.WebJobs.Host/Common/GenericBinder.cs

```diff
@@ -93,19 +115,40 @@ internal class GenericBinder
             }         
         }
 
-        // Bind an IAsyncCollector<TMessage> to a user parameter. 
-        // This handles the various flavors of parameter types and will morph them to the connector.
-        // parameter  - parameter being bound. 
-        // TContext - helper object to pass to the binding the configuration state. This can point back to context like secrets, configuration, etc.
-        // builder - function to create a new instance of the underlying Collector object to pass to the parameter. 
-        //          This binder will wrap that in any adpaters to make it match the requested parameter type.
+        /// <summary>
+        /// Bind a  parameter to an IAsyncCollector. Use this for things that have discrete output items (like sending messages or writing table rows)
+        /// This handles the various flavors of parameter types and will morph them to the connector.
+        /// </summary>
+        /// <typeparam name="TMessage">The native type of message. For example, for azure queues, this would be CloudQueueMessage.</typeparam>
+        /// <typeparam name="TContext">helper object to pass to the binding the configuration state. This can point back to context like secrets, configuration, etc.</typeparam>
+        /// <param name="parameter">parameter being bound.</param>
+        /// <param name="converterManager"></param>
+        /// <param name="builder">function to create a new instance of the underlying Collector object to pass to the parameter. 
+        /// This binder will wrap that in any adpaters to make it match the requested parameter type.</param>
+        /// <param name="invokeString">a short string summary that describes this binding. That will display on the dashboard</param>
+        /// <param name="invokeStringBinder">retrieve a context from the invoke string. </param>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "string")]
         public static IBinding BindCollector<TMessage, TContext>(
```

Comment:
It comes out of GenericBinder.  GenericBinder may be calling into the new IConverterMAnager to realize it can't make the binding. 

---
In reply to: [54933245](https://github.com/Azure/azure-webjobs-sdk/pull/643#discussion_r54933245) [](ancestors = 54933245)